RouteDescriber/FosRestDescriber.php

```diff
@@ -104,6 +105,17 @@ private function getPattern($requirements)
     private function getFormat($requirements)
     {
         if ($requirements instanceof Constraint && !$requirements instanceof Regex) {
+            if ($requirements instanceof DateTime) {
+                // As defined per RFC3339
+                if (\DateTime::RFC3339 === $requirements->format || 'c' === $requirements->format) {
+                    return 'date-time';
+                } elseif ('Y-m-d' === $requirements->format) {
+                    return 'date';
+                } else {
+                    return null;
+                }
```

Comment:
```suggestion
                }
                
                if ('Y-m-d' === $requirements->format) {
                    return 'date';
                }

                return null;
```
BTW, I think we should keep the existing behavior if the provided format is not supported or throw an exception in that case.

Reply:
Whether we add a ``format: DateTime`` or not at all in the resulting OA doc, the Rest Client Generator (.net) will create a string parameter as the DateTime format is unknown. So for me it should not appear as not supported by https://swagger.io/specification/. I'm not for throwing an exception as if someone wants to provide an other PHP supported datetime format it should be possible but just ignored by Open API.

Reply:
Throwing an exception would be too bad for DX imo, ignoring the constraint in case we don't know what to do looks like a fine compromise to me.